### ISSUE TITLE
Unable to set Hostname for eureka registered service on cloud

### ISSUE THREAD

**Issue by ghost** [spring-cloud/spring-cloud-netflix#1558]
Hi,
I have a eureka server and spring boot admin server running in spring pivotal cloud. I have a application which have configuration related to the eureka server as well as spring boot admin. Attaching the project for reference. Spring boot version: 1.4.2.RELEASE, Spring cloud version: Brixton.SR4.
Problem is the service is getting registered with the random name and also the spring boot admin is not able to fetch the actuator endpoints. I have tried with the hostname, preferIpAddress, healthCheckUrl in eureka instance configuration but to no effect. Also attaching the snapshots for the same. Locally everything works fine but when deployed on cloud we are facing the below issues.

![eureka_server](https://cloud.githubusercontent.com/assets/20604128/21261977/b8c73da0-c3b4-11e6-81b4-4cb2407b09f0.PNG)

![spring-admin](https://cloud.githubusercontent.com/assets/20604128/21261984/c1be7e28-c3b4-11e6-8dc0-8f310597d7e4.PNG)

![demoapp_logs](https://cloud.githubusercontent.com/assets/20604128/21261990/cb756c6a-c3b4-11e6-8307-a029c51cc0f8.PNG)

Spring boot admin logs:
![spring-admin_logs](https://cloud.githubusercontent.com/assets/20604128/21261698/0cbc262a-c3b3-11e6-80b6-468c58dbcbb1.PNG)

[Demoapp.zip](https://github.com/spring-cloud/spring-cloud-netflix/files/657084/Demoapp.zip)

**Comment by ryanjbaxter** [comment]
If you hover over the link in the Eureka dashboard (or inspect it in the browser) what is the URL it links to?  The UI just displays the instance ID, the URL that instance registered with might be completely different.  I deployed you sample to my org on run.pivotal.io and it registered with my Eureka instance with the correct URL.  The only change I had to make was to add `eureka.instance.nonSecurePort = 80` in `application.yml`.
